packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js

```diff
@@ -48,4 +49,16 @@ export class GETSurveyResponses extends GETHandler {
     // Apply regular permissions
     return this.getPermissionsFilter(dbConditions, options);
   }
+
+  async countRecords(criteria) {
+    // Only join tables that we are filtering on
+    const { multiJoin } = getQueryOptionsForColumns(
```

Comment:
The base `countRecords()` joins all the tables required in the columns. However joining on the `entity` and `survey` tables takes the query time up to ~8 seconds (compared to < 1 second without joining).

It does mean however that this number _could_ be higher than the actual row count returned by this route, as it's possible there are survey responses with no reference entities. We could put a constraint against that in the database? Or we could just leave this as possibly saying there are more records than will appear in the Admin Panel (which isn't going to break anything)

Reply:
No reference entities means no entity name showing up in admin panel?

Reply:
No, this is just the `count` part, so it basically means that we'll count survey responses as existing even if they have no related entity 👍

Reply:
Ah cool, that should be ok then.